flight/Modules/Attitude/attitude.c

```diff
@@ -1015,7 +1018,7 @@ static int32_t updateAttitudeINSGPS(bool first_run, bool outdoor_mode)
 		GPSVelocityGet(&gpsVelData);
 
 	// Discard mag if it has NAN (normally from bad calibration)
-	mag_updated &= (magData.x == magData.x && magData.y == magData.y && magData.z == magData.z);
+	mag_updated &= !(IS_NOT_FINITE(magData.x) || IS_NOT_FINITE(magData.y) || IS_NOT_FINITE(magData.z));
```

Comment:
I struggled with that one because it's uglier than before. In retrospect, I think I should have made an ```IS_FINITE``` macro which simply inverts IS_NOT_FINITE.